src/vcpkg/sourceparagraph.cpp

```diff
@@ -1544,7 +1544,18 @@ namespace vcpkg
 
     static bool is_dependency_trivial(const Dependency& dep)
     {
-        return dep.features.empty() && dep.platform.is_empty() && dep.extra_info.is_empty() &&
+        if (Util::Vectors::contains(dep.features, "default"))
```

Comment:
I believe triviality now depends on the `depend-defaults` setting, because it changes the definition of `"zlib"` to imply `"default-features": false`.

That means that `"default-features": true` is no longer trivial if `"depend-defaults": true`.

Reply:
In general this is true, but I think that this is not possible at the moment. A SourceControlFile is afaik only created by parsing a file currently. To set `default-features` to something different than `depend-defaults` you have to set `default-feautes`. This is saved in `Optional<bool> dep.default_features` and a dependency is non trivial when the optional has a value.  
But I can change this. This would then remove unnecessary `"default-features": true` when `depends-default` is true (the default)

Reply:
But this is then wrong when using CONTROL files because there the `features` field has no entry `default` iiuc. 

Imho we should get rid off "default" and "core" in the features vector and make default_features a bool again or remove the default_features field completely and always add add "default" and "core" to the features vector. But currently this is a mess.

Reply:
I think I will leave it as it is in this PR because it is working right now.

Reply:
This would be easy to solve when https://github.com/microsoft/vcpkg-tool/pull/813 gets merged

Reply:
> But currently this is a mess.

If this is true, then nothing to argue. 🙃